doc/self-hosted/build.md

```diff
@@ -0,0 +1,45 @@
+# Building instructions for code-server 🏗️
```

Comment:
I'd change this to just `Building`. `code-server` is already expected since they are in the repo